domehunter/__init__.py

```diff
@@ -86,70 +230,291 @@ def is_home(self):
+        """Calibrate the encoder (determine degrees per tick)"""
+        # rotate the dome until we hit home, to give reference point
+        self._move_cw()
+        self.home_sensor.wait_for_active(timeout=self.wait_timeout)
+        if self.testing:
+            # in testing mode we need to "fake" the activation of the home pin
+            self.home_sensor_pin.drive_high()
+        time.sleep(0.1)
+        self._stop_moving()
+        self.encoder_count = 0
 
-    def halt_daemon(self):
-        """Maybe start the RCP daemon here?."""
-        raise NotImplementedError
+        # now set dome to rotate n times so we can determine the number of
+        # ticks per revolution
+        rotation_count = 0
+        while rotation_count < num_cal_rotations:
+            time.sleep(0.5)
+            self._move_cw()
+            if self.testing:
+                # tell the fake home sensor that we have left home
+                self.home_sensor_pin.drive_low()
+                self._simulate_ticks(num_ticks=10)
+            self.home_sensor.wait_for_active(timeout=self.wait_timeout)
+            if self.testing:
+                # tell the fake home sensor that we have come back to home
+                self.home_sensor_pin.drive_high()
+            time.sleep(0.5)
+            self._stop_moving()
+
+            rotation_count += 1
+
+        # set the azimuth per encoder tick factor based on how many ticks we
+        # counted over n rotations
+        self.az_per_tick = 360 / (self.encoder_count / rotation_count)
+        pass
 
-##################################################################################################
+###############################################################################
 # Private Methods
-##################################################################################################
+###############################################################################
 
     def _set_at_home(self):
+        """Update home status to at home and debug LEDs (if enabled)."""
+        self._turn_led_on(leds=['input_2'])
         self._at_home = True
 
     def _set_not_home(self):
+        """Update home status to not at home and debug LEDs (if enabled)."""
+        self._turn_led_off(leds=['input_2'])
         self._at_home = False
 
-    def _increment_count(self, device=None):
-        print(f"{device} activated _increment_count")
+    def _increment_count(self):
+        """
+        Private method used for callback function of the encoder DOD.
+
+        Calling this method will toggle the encoder debug LED (if enabled)
+        and increment or decrement the encoder_count instance variable,
+        depending on the current rotation direction of the dome.
+
+        If the current dome direction cannot be determined, the last recorded
+        direction is adopted.
+        """
+        print(f"Encoder activated _increment_count")
+        self._turn_led_on(leds=['input_1'])
+        time.sleep(0.01)
```

Comment:
you could build in a `duration=0.01` into definition of `self._turn_led_on` as you must want to do this often and so you don't have to do `self._turn_led_off(leds=['input_1'])` later on?